src/event_bus/event_bus.py

```diff
@@ -0,0 +1,202 @@
+        """Subscribe to an event type
+
+        :param callback: A callable that gets called when the event is triggered
+        :param event_type: The type of the event
+        :return: A handle to this subscription
+        """
+        handle = uuid4()
+        self.callbacks[handle] = (callback, event_type)
+        return handle
+
+    def unregister_callback(self, handle: UUID):
+        """Cancels the subscription associated to a handle
+
+        :param handle: the handle
+        """
+        self.callbacks.pop(handle)
+
+    def _get_callbacks_for_event_type(
+        self, event_type: EventType
+    ) -> Iterator[Callable[[Event], None]]:
+        """Returns an iterator to all callbacks associated with th eenevent type
+
+        :param event_type: The event type
+        :return: The iterator to all the callbacks
+        :yield: a callback
+        """
+        return (
+            callback
+            for callback, e_type in self.callbacks.values()
+            if e_type == event_type
+        )
+
+    def _build_arguments(
+        self, name: str, arg_keys: list[str], args: list, kwargs: dict
+    ) -> dict:
+        """Constructs an argument dict from a list of keys that have to be included
+        a list of argument values and an kwargs-dict
+
+        :param arg_keys: the list of keys
+        :param args: the list of values
+        :param kwargs: the kwargs dict
+        :return: the argument dict
+        """
+        if not len(args) + len(kwargs.keys()) == len(arg_keys):
+            raise TypeError(
+                f"{self.__class__.__name__}.{name}() takes exactly {len(arg_keys)} "
+                f"arguments ({len(args) + len(kwargs.keys())} given)"
+            )
+        arguments = dict(zip(arg_keys, args))
+
+        arg_keys_set = set(key for key in arg_keys if key not in arguments)
+        kwarg_keys_set = set(kwargs.keys())
+        for arg_key in arg_keys_set:
+            if arg_key not in kwarg_keys_set:
+                raise TypeError(
+                    f"{self.__class__.__name__}.{name}() missing required argument '{arg_key}'"
+                )
+        for kwarg_key in kwarg_keys_set:
+            if kwarg_key not in arg_keys_set:
+                raise TypeError(
+                    f"{self.__class__.__name__}.{name}() got an unexpected "
+                    f"keyword argument '{kwarg_key}'"
+                )
+
+        return arguments | kwargs
+
+    def __getattr__(self, name: str) -> object:
+        """This pretends to a caller that the class has the methods
+        defined in EVENT_METHODS. It returns a callable that creates
+        an event and calls all callbacks.
+
+        :param name: the method name
+        :raises AttributeError: when unknown method name is passed
+        :return: a callable
+        """
+        if name not in self.EVENT_METHODS:
+            raise AttributeError(
+                f"'{self.__class__.__name__}' object has no attribute '{name}'"
```

Comment:
Why does self. __class__. __name__ not always refer to EventBus? If it does, I think you should make it explicit.

Reply:
This version minimizes the amount of places where you would have to change code when you would rename the class. It therefore reduces redundancy. But if it really bugs you I can change it and rely on modern IDE magic for renaming :)

Reply:
No this makes sense to me. God job